pkg/component/devfile/adapters/kubernetes/kubernetes.go

```diff
@@ -0,0 +1,68 @@
+package kubernetes
+
+import (
+	"github.com/golang/glog"
+	"github.com/openshift/odo/pkg/component/devfile/adapters/common"
+	devfileCommon "github.com/openshift/odo/pkg/devfile/versions/common"
+	"github.com/openshift/odo/pkg/kclient"
+	"github.com/openshift/odo/pkg/log"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// Adapter maps Devfiles to Kubernetes resources and actions
+type Adapter struct {
+	common.DevfileAdapter
+	Client kclient.Client
+}
+
+// Start creates Kubernetes resources that correspond to the devfile if they don't already exist
+func (k Adapter) Start() error {
+
+	err := k.Devfile.Data.Validate()
+	if err != nil {
+		return errors.Wrap(err, "The devfile is invalid")
+	}
+
+	componentName := k.DevfileComp.Name
+
+	if k.componentExists(componentName) {
+		log.Info("The component already exists")
+		return nil
+	}
+
+	var containers []corev1.Container
+	// Only components with aliases are considered because without an alias commands cannot reference them
+	for _, comp := range k.Devfile.Data.GetAliasedComponents() {
+		if comp.Type == devfileCommon.DevfileComponentTypeDockerimage {
+			glog.V(3).Infof("Found component %v with alias %v\n", comp.Type, *comp.Alias)
+			envVars := convertEnvs(comp.Env)
+			resourceReqs := getResourceReqs(comp)
```

Comment:
Can we pass `&comp.MemoryLimit` to the function instead of the entire `comp` to make it consistent like the other util functions?

Reply:
I'd prefer to leave this as is because it's likely we'll handle other resource requirements in the future.